orderer/common/cluster/block_fetcher_test.go

```diff
@@ -0,0 +1,65 @@
+package cluster_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	"github.com/hyperledger/fabric/orderer/common/cluster/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+/*
+ Scenarios to test
+
+1. Two orderers, first one withold blocks. BlockFetcher should try pulling from the first one, it should time out and
+   then pull from the second orderer.
+2. Happy Path, one orderer, block fetcher fetches blocks from it.
+
+*/
+func TestBlockFetcherHappyPath(t *testing.T) {
+	bs := &mocks.BlockSource{}
+
+	bs.On("PullBlock", mock.Anything).Return(&common.Block{
+		Header:               &common.BlockHeader{Number: 1},
+		Data:                 &common.BlockData{},
+		Metadata:             &common.BlockMetadata{},
+		XXX_NoUnkeyedLiteral: struct{}{},
+		XXX_unrecognized:     []byte{},
+		XXX_sizecache:        0,
+	})
+
+	bs.On("UpdateEndpoints", mock.Anything)
+	bf := cluster.NewBlockFetcher(bs)
+	bf.ShuffleTimeout = time.Second * 1
+	bf.Logger = flogging.MustGetLogger("test")
+
+	// add 2 endpoints to pull from
+	bf.Endpoints = []cluster.EndpointCriteria{{Endpoint: "localhost:5100"}, {Endpoint: "localhost:5200"}}
+
+	// pullblock from any endpoint
+	require.Equal(t, uint64(1), bf.PullBlock(1).Header.Number)
+}
+
+func TestBlockFetcherBlockWithold(t *testing.T) {
+	bs := &mocks.BlockSource{}
+
+	bs.On("PullBlock", mock.Anything).Return(&common.Block{
+		Header:               &common.BlockHeader{Number: 1},
+		Data:                 &common.BlockData{},
+		Metadata:             &common.BlockMetadata{},
+		XXX_NoUnkeyedLiteral: struct{}{},
```

Comment:
no need to put these protobuf fields (XXX)

Reply:
ok